tests/CMakeLists.txt

```diff
@@ -44,12 +44,21 @@ endforeach(cpp_test_file)
 
 ### C tests ###
 if(BUILD_C)
-    link_directories( ${CMAKE_SOURCE_DIR}/bin )
-
-    add_executable ( test_runner_c test_runners/test_runner.c)
-    target_link_libraries(test_runner_c co_sim_io_c)
-
-    install( TARGETS test_runner_c DESTINATION ${CMAKE_SOURCE_DIR}/bin )
+    function(add_c_test TEST_SOURCE_FILE)
+        get_filename_component(TEST_FILENAME ${TEST_SOURCE_FILE} NAME)
+        string(REGEX REPLACE "[.]c" "_c_test" TEST_NAME ${TEST_FILENAME})
+        message(STATUS  "adding ${TEST_NAME} ")
+        add_executable(${TEST_NAME} ${TEST_SOURCE_FILE})
+        target_link_libraries(${TEST_NAME} co_sim_io_c)
+        add_test(${TEST_NAME} ${TEST_NAME})
```

Comment:
As you may have observed in my latest change, I am adding the tests manually to be able to run them together, like import, export

Reply:
I understand what you mean
I this case adding the tests directly is what we need, since those are the standalone tests

The tests for the connection etc (like you are doing now) will come in a different PR, then we need the function you describe